src/Skia/Avalonia.Skia/Gpu/OpenGl/GlSkiaExternalObjectsFeature.cs

```diff
@@ -109,63 +116,63 @@ SKColorType ConvertColorType(PlatformGraphicsExternalImageFormat format) =>
             _ => SKColorType.Rgba8888
         };
 
-    SKSurface? TryCreateSurface(int target, int textureId, int format, int width, int height, bool topLeft)
+    SKImage? TryCreateImage(int target, int textureId, int format, int width, int height, bool topLeft)
     {
-        var origin = topLeft ? GRSurfaceOrigin.TopLeft : GRSurfaceOrigin.BottomLeft; 
+        var origin = topLeft ? GRSurfaceOrigin.TopLeft : GRSurfaceOrigin.BottomLeft;
+
         using var texture = new GRBackendTexture(width, height, false,
             new GRGlTextureInfo((uint)target, (uint)textureId, (uint)format));
-        var surf = SKSurface.Create(_gpu.GrContext, texture, origin, SKColorType.Rgba8888);
-        if (surf != null)
-            return surf;
-        
+
+        var image = SKImage.FromAdoptedTexture(_gpu.GrContext, texture, origin, SKColorType.Rgba8888);
+        if (image is not null)
+            return image;
+
         using var unformatted = new GRBackendTexture(width, height, false,
-            new GRGlTextureInfo((uint)GlConsts.GL_TEXTURE_2D, (uint)textureId));
-        
-        return SKSurface.Create(_gpu.GrContext, unformatted, origin, SKColorType.Rgba8888);
+            new GRGlTextureInfo((uint)target, (uint)textureId));
+
+        return SKImage.FromAdoptedTexture(_gpu.GrContext, unformatted, origin, SKColorType.Rgba8888);
     }
-    
+
     IBitmapImpl TakeSnapshot()
     {
         var width = _image?.Properties.Width ?? _sharedTexture!.Size.Width;
         var height = _image?.Properties.Height ?? _sharedTexture!.Size.Height;
         var internalFormat = _image?.InternalFormat ?? _sharedTexture!.InternalFormat;
         var textureId = _image?.TextureId ?? _sharedTexture!.TextureId;
         var topLeft = _image?.Properties.TopLeftOrigin ?? false;
-        var textureType = _image?.TextureType ?? GlConsts.GL_TEXTURE_2D;
-        
-        
-        IBitmapImpl rv;
-        using (var surf = TryCreateSurface(textureType, textureId, internalFormat, width, height, topLeft))
+        var textureType = _image?.TextureType ?? GL_TEXTURE_2D;
+
+        var snapshotTextureId = CopyToNewTexture(textureType, textureId, internalFormat, width, height);
+        var snapshotImage = TryCreateImage(textureType, snapshotTextureId, internalFormat, width, height, topLeft);
+
+        if (snapshotImage is null)
+            throw new OpenGlException("Unable to consume provided texture");
```

Comment:
Need to delete `snapshotTextureId` here since texture ownership wasn't transferred to Skia